### PR TITLE
:lipstick: [#1795] Show selected documents to be uploaded

### PR DESCRIPTION
**Pull request by jiromaykin**
issue: https://taiga.maykinmedia.nl/project/open-inwoner/task/1795
For now: implementing for Single upload only; so is mostly a styling PR, with added logic that wil make it easier to append HTML later on, when this become a Multiple upload file input.
 ToDo:

- [x] Add large spacing around file input button, change icon
- [x] change javascript classes into appended HTML template for each uploaded file (1): a column with document icon, file-name, file-extension, file size in MB, icon with trashcan.
- [x] Move selected file info (name/size) below upload button, instead of above.
- [x] add delete button that will remove 1 file from array
- [x] check to see if post still works when using an array instead of pure fileList object.
- [x] move documents upload requirements info below the select button
- [x] keep documents-list of already uploaded files _below_ upload section

Tasks with Sven:
- [x] make disabled/enabled button 100% wide on mobile, remove icon + move to CaseDetail.SCSS
- [x] Show button as disabled when input is empty
- [x] remove unused JS  files in upload-document JS folder
- [x] put global vanilla styling for other input-type=file back into separate file (FileInputDefault)
- [x] check padding/margins
- [x] check Tests

Codepen sketch: https://codepen.io/jirosworld/pen/oNJwOaq

### CODE REVIEW

**Comment by svenvandescheur** [comment]
I've updated this PR to implement a file picker:

 - That support drag and drop
 - Is technically prepared for multiple file uploads
 - Shows selected file(s) with file name, extension and file size.
 - Is. based on the original design.

Scherm­afbeelding 2023-11-17 om 15 03 19

Currently multiple files upload is not yet possible (although the frontend component can be made capable by setting the multiple attribute on the input and changing the text labels accordingly). I'm unsure whether the backend (and upstream) service are capable of dealing with multiple file uploads. Another ticket (#1974) tracks this progress.

File: src/open_inwoner/components/templates/components/Button/Button.html [hunk 1]
```diff
@@ -21,7 +21,7 @@
         {% endif %}
     </a>
 {% else %}
-    <button
+    <{{ tag }}
```
[the remark this answers is not in the document]
Comment (svenvandescheur) [reply src/open_inwoner/components/templates/components/Button/Button.html:24]: Check, in dat geval moet in de input zelf waarschijnlijk ook het een en andere geinlined worden.

**Comment by pi-sigma** [comment]
Needs rebasing on `develop`

File: src/open_inwoner/cms/cases/tests/test_htmx.py [hunk 2]
```diff
@@ -441,13 +442,9 @@ def test_cases(self, m):
                 }
             ],
         )
-        submit_button = upload_form.get_by_role("button", name=_("Document uploaden"))
+        submit_button = upload_form.get_by_role("button", name=_("Upload documenten"))
```
Comment (pi-sigma) [inline src/open_inwoner/cms/cases/tests/test_htmx.py:445]: This seems to be assuming that the user can upload multiple documents, contrary to the description of the ticket. Is this intended?

**Review by pi-sigma — DISMISSED** [review]

File: src/open_inwoner/cms/cases/tests/test_htmx.py (hunk 2 above)
Comment (svenvandescheur) [reply src/open_inwoner/cms/cases/tests/test_htmx.py:445]: I'm leaving this as I think it's a minor issue and will be correct once multiple support has landed.

File: src/open_inwoner/cms/cases/forms.py [hunk 3]
```diff
@@ -23,8 +20,19 @@ class CaseUploadForm(forms.Form):
...
+        self.oz_config = OpenZaakConfig.get_solo()
         super().__init__(**kwargs)
 
+        help_text = f"Grootte max. { self.oz_config.max_upload_size } MB, toegestane document formaten: { ', '.join(self.oz_config.allowed_file_extensions) }."
+
+        try:
+            ztc = ZaakTypeConfig.objects.filter_case_type(case.zaaktype).get()
+            help_text = ztc.description or help_text
+        except (AttributeError, ObjectDoesNotExist):
+            pass
```
[the remark this answers is not in the document]
Comment (svenvandescheur) [reply src/open_inwoner/cms/cases/forms.py:32]: The AttributeError occurred in `case.zaaktype` during the test, I've re-added the AttributeError but feel free to fix this later.

**Review by svenvandescheur — APPROVED** [review]

**Pull request merged**